packages/search-ui-app-search-connector/src/__tests__/AppSearchAPIConnector.test.js

```diff
@@ -207,7 +208,7 @@ describe("AppSearchAPIConnector", () => {
         filters: {
           all: [
             {
-              all: [
+              any: [
```

Comment:
I'm worried that the meaning of `type` has changed in our `Filter` model.

That parameter previously affected the inner filter, but now there is an `innerFilter` which affects that. This is apparent by looking at this test.

I think you are using the term "innerType" to refer to nested filters:

```
filters: {
  all: [{ // type
    any: {} // innerType
  }]
}
```

Because `type` is already used for nested filters, I think we should update the semantics to this:

```
filters: {
  all: [{ // outertype
    any: {} // type
  }]
}
```

This way we can safely add a `outerType` property without a breaking change, and leave the meaning of `type` in place.

If we do go that route, this test would also need tests to test the `outerType`, whether it's here or in `requestAdapter.test.js`